pkg/controller/cnp/clusternodeprediction_controller.go

```diff
@@ -0,0 +1,114 @@
+package cnp
+
+import (
+	"context"
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/version"
+	"k8s.io/client-go/scale"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	predictionapi "github.com/gocrane/api/prediction/v1alpha1"
+)
+
+type ClusterNodePredictionController struct {
+	client.Client
+	Log         logr.Logger
+	Scheme      *runtime.Scheme
+	RestMapper  meta.RESTMapper
+	Recorder    record.EventRecorder
+	scaleClient scale.ScalesGetter
+	K8SVersion  *version.Version
+}
+
+func (c *ClusterNodePredictionController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	// Log for controller information
+	c.Log.Info("got", "cnp", req.NamespacedName)
```

Comment:
cloud you please more logs/event. for main code branches (and set log level to 4 for debug info if there is any)